tasks/custom_task/custom_task_test.py

```diff
@@ -0,0 +1,23 @@
+import inspect
+from invoke import Task
+import invoke
+import unittest
+
+expected_content = """    def __call__(self, *args: Any, **kwargs: Any) -> T:
+        # Guard against calling tasks with no context.
+        if not isinstance(args[0], Context):
+            err = "Task expected a Context as its first arg, got {} instead!"
+            # TODO: raise a custom subclass _of_ TypeError instead
+            raise TypeError(err.format(type(args[0])))
+        result = self.body(*args, **kwargs)
+        self.times_called += 1
+        return result
+"""
+
+
+class TestTaskCall(unittest.TestCase):
+    def test_task_call_content(self):
+        self.assertEqual(inspect.getsource(Task.__call__), expected_content)
```

Comment:
I do not think this test is really relevant. It seems hard to maintain, for example we will need to modify the test when TODO is completed.
To me if you want to test that this custom call is working you should try to find a way to mock the logger and to verify that a log is actually emitted when a task is called

Reply:
You're right ! The unit test should just test that the custom call saves the right log.

I used the unit test here because I wanted a way to check that the `custom__call__` function was adapted with the `invoke` version we're using. I need to find another way of doing this.

Reply:
This is now fixed !